### PR TITLE
e2e test for cluster config change - GITOPS-1003

### PR DESCRIPTION
**Pull request by ishitasequeira**
What type of PR is this?
/kind enhancement

What does this PR do / why we need it:
This PR add the following test case.

Create an argocd application CR that has source folder containing a manifest of an config map in openshift-config project
oc apply -f the argocd application CR
Verify that the openshift cluster-config resource is deployed. Verify that the argocd application is healthy and sync'ed.

Have you updated the necessary documentation?

 Documentation update is required by this PR.
 Documentation has been updated.
Which issue(s) this PR fixes:

Fixes - https://issues.redhat.com/browse/GITOPS-1003

Test acceptance criteria:

 Unit Test - NA
 E2E Test
How to test changes / Special notes to the reviewer:

### CODE REVIEW

File: test/e2e/gitops_service_test.go [hunk 1]
```diff
@@ -301,3 +303,38 @@ func validateMachineConfigUpdates(t *testing.T) {
...
+		t.Fatal(err)
+	}
+
+	cmd := exec.Command(ocPath, "apply", "-f", schedulerYAML)
+	err = cmd.Run()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	time.Sleep(5 * time.Second)
```
Comment (jgwest) [inline test/e2e/gitops_service_test.go:325]: We probably don't want to add more `time.Sleep(...)` calls if we can help it. See https://github.com/redhat-developer/gitops-operator/issues/130 for outline of alternatives. 

You should be able to do something like this (pseudocode):
```go

// import this:
//	"k8s.io/apimachinery/pkg/util/wait"


err := wait.Poll(time.Second*1, time.Second*60, func() (bool, error) {
	
	if helper.ApplicationHealthStatus("scheduler", "openshift-gitops"); err != nil {
		// print the error received
		return false, nil
	}

	if helper.ApplicationSyncStatus("scheduler", "openshift-gitops"); err != nil {
		// print the error received	
		return false, nil
	}
	
	existingConfigMap := &corev1.ConfigMap{}
	err = f.Client.Get(context.TODO(), types.NamespacedName{Name: "policy-configmap", Namespace: "openshift-config"}, existingConfigMap)
	if err != nil {
			// print the error received
			return false, nil
	}

		return true, nil
})
```

Comment (ishitasequeira) [reply test/e2e/gitops_service_test.go:325]: I thought Amit was going to handle that in a separate task. But I can surely update this test.

**Comment by amitkrout** [comment]
@ishitasequeira Can you please rebase.

File: test/helper/application_status.go [hunk 2]
```diff
@@ -7,6 +7,25 @@ import (
...
+// ProjectExists return true if the AppProject exists in the namespace,
+// false otherwise (with an error, if available).
+func ProjectExists(projectName string, namespace string) (bool, error) {
+	var stdout, stderr bytes.Buffer
+	ocPath, err := exec.LookPath("oc")
+	if err != nil {
+		return false, err
+	}
+
+	cmd := exec.Command(ocPath, "get", "appproject/"+projectName, "-n", namespace, "-o", "yaml")
```
Comment (amitkrout) [inline test/helper/application_status.go:19]: I think arguments ```"-o", "yaml"``` are not needed

File: test/helper/application_status.go [hunk 3]
```diff
@@ -7,6 +7,25 @@ import (
...
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	if err := cmd.Run(); err != nil {
+		return false, fmt.Errorf("oc command failed: %s%s", stdout.String(), stderr.String())
```
Comment (amitkrout) [inline test/helper/application_status.go:23]: Can we make it more expressive like 
```return false, fmt.Errorf("oc command failed. Stdout: %s, Stderr: %s", stdout.String(), stderr.String())```

**Review by amitkrout — CHANGES_REQUESTED** [review]

**Comment by amitkrout** [comment]
looks good to me. However i seek someone's review from the team. ping @wtam2018 
/approve

**Pull request merged**